### ISSUE TITLE
JSDoc optional argument does not generate an error in strict mode

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#35485]
from https://github.com/microsoft/vscode/issues/85015

**TypeScript Version:**  3.8.0-dev.20191203

**Search Terms:**

**Code**
In a jsconfig project:

```json
{
    "compilerOptions": {
        "strict": true,
        "checkJs": true
    },
    "exclude": [
        "node_modules",
        "**/node_modules/*"
    ]
}
```

```js
// @ts-check

/**
 * @param {object} adventurer Information about the adventurer.
 * @param {string} [adventurer.dog] The adventurer's dog.
 */
export function test(adventurer) {
	return adventurer.dog.split(' ');
}
```

**Expected behavior:**
There should be an error on the call to split since `dog` is potential undefined.

**Actual behavior:**
No error

**Playground Link:** 

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
This is fixed. With the supplied `strict` and `checkJs` project, `3.8.0-dev.20191203` produces no error for `adventurer.dog.split(' ')`; `4.0.0-dev.20200708` is also affected, while `4.0.0-dev.20200709` reports TS2532, `Object is possibly 'undefined'`, at that expression. Current native TypeScript reports the equivalent TS18048 diagnostic.

The change is [#39487](https://github.com/microsoft/TypeScript/pull/39487), which corrected optional JSDoc `@param` and `@property` handling.
